### PR TITLE
Introduce kubernetes package granularity

### PR DESCRIPTION
**Pull request by ChrsMark**
## What does this PR do?
Changes `kubernetes` package to use input groups in order to add icons for each service to support granular search.
This PR follows approach used in https://github.com/elastic/integrations/pull/767.
This package follows the structure suggested at https://github.com/elastic/integrations/issues/1017#issuecomment-811824576.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

## How to test this PR locally
1. Search for the integration in integrations search in Kibana
2. Check that all tiles are present
3. Select each tile and verify content is as expected.

## Related issues
- Closes https://github.com/elastic/integrations/issues/1017

## Screenshots
Screenshot 2021-05-27 at 3 38 55 PM

### CODE REVIEW

File: packages/kubernetes/manifest.yml [hunk 1]
```diff
@@ -16,30 +16,167 @@ screenshots:
...
+    data_streams:
+      - container
+      - node
+      - pod
+      - system
+      - volume
+    inputs:
+      - type: kubernetes/metrics
+        title: Collect Kubernetes metrics from Kubelet API
+        description: Collecting Node, Pod, Container, Volume and System metrics from Kubelet
```
Comment (sorantis) [inline packages/kubernetes/manifest.yml:37]: @ChrsMark @mtojek folks, is input description visible anywhere in the UI?

Comment (ChrsMark) [reply packages/kubernetes/manifest.yml:37]: Hm, I'm not sure, tbh I have not yet tested this one manually and I don't think I can until https://github.com/elastic/kibana/pull/99866 is merged. @kaiyan-sheng does is it happen you know more about this?

Comment (jen-huang) [reply packages/kubernetes/manifest.yml:37]: elastic/kibana#99866 is merged but that doesn't include changes to the policy editor. currently the `description` field of inputs is not surfaced in the UI (instead we use `title` next to the enable/disable input toggle) and I don't foresee the upcoming changes to policy editor to change that. @sorantis if we think it is import to surface description, maybe a quick win would be to add an info icon with a tooltip

**Comment by ChrsMark** [comment]
I gave it a shot. I see that search page looks good:
Screenshot 2021-05-27 at 3 38 55 PM

However I see some inconsistencies when I open a “tab” for example kube-state-metrics:

1. Docs for metricsets/data_streams are missing and I only see the basic docs
Screenshot 2021-05-27 at 3 46 47 PM
2. The description of the integration in settings view is the one of kubelet  input_group
Screenshot 2021-05-27 at 3 46 23 PM

Last but not least data_streams of the input_group selected are not enabled by default. Only data_streams that are defined as “enabled by default” in package are enabled. Not sure if this is proper UX.

cc: @sorantis @kaiyan-sheng @jen-huang 

Note: I have not managed to manually test the integration itself to check if metrics are being collected since I'm still trying to enrol an agent running on k8s so as to test it properly.

**Comment by sorantis** [comment]
@ChrsMark looks good! Nit (aligning with the official names https://kubernetes.io/docs/concepts/overview/components/): 
- Kube-state-metrics -> kube-state-metrics
- Api-server -> kube-apiserver
- Proxy -> kube-proxy
- Controller-manager -> kube-controller-manager
- Scheduler -> kube-scheduler
- Event -> Events

What's the difference between Kubelet and Kubernetes tiles?

**Comment by ChrsMark** [comment]
Screenshot 2021-05-27 at 5 42 13 PM
Updated the naming ^^^


> What's the difference between Kubelet and Kubernetes tiles?
@sorantis could you elaborate more on this please? Under `policy_templates` we have 7 input_groups but UI apparently adds one extra tile (Kubernetes) for the whole integration I guess?

**Comment by sorantis** [comment]
Is it possible to hide either one of them? I recall a conversation where we discussed the possibility to control what tiles we show to the user. cc @jen-huang is it supported?

**Comment by kaiyan-sheng** [comment]
> 1. Docs for metricsets/data_streams are missing and I only see the basic docs

@ChrsMark Are you talking about documentation for kubelet, apiserver and etc(for example https://github.com/elastic/integrations/pull/1018/files#diff-3dc8f2b9506ab2ebd53667776cce858fb22dbf740deb62c9687a4da4fc9a1a92)? These are under their own sub tiles once you click the icon of kubelet or apiserver, you can see the documentation there.

**Comment by ChrsMark** [comment]
With @kaiyan-sheng we figured out the issue related to the docs (reported above). Fixed with https://github.com/elastic/integrations/pull/1018/commits/a7ea2fc720e620d8fbcc59a806af0889511cc10a. Long story short we group docs's files by input_group, hence we want 5 docs' files (`.md`) one per input_group. Various data_streams' docs are group under its respective input_group's docs file.

**Comment by ChrsMark** [comment]
So what is not addressed yet is to only show data_streams that belong to selected input_group's tile. For example if I select kube-state-metrics I should only see settings for state_* data_streams enabled. 

@jen-huang @sorantis looking into https://github.com/elastic/kibana/issues/93315 it should be covered by item4 of the list but I'm just raising it here for awareness. 

If we cover the above shall we completely remove https://github.com/elastic/integrations/blob/master/packages/kubernetes/data_stream/state_container/manifest.yml#L6?

**Comment by jen-huang** [comment]
> So what is not addressed yet is to only show data_streams that belong to selected input_group's tile. For example if I select kube-state-metrics I should only see settings for state_* data_streams enabled.

Yeah, this is not implemented yet as policy editor changes hasn't landed yet (https://github.com/elastic/integrations/pull/1018#discussion_r639986956). Hold tight :)

> Is it possible to hide either one of them? I recall a conversation where we discussed the possibility to control what tiles we show to the user. cc @jen-huang is it supported?

IIRC from our discussions we decided to always show the base package tile, plus all the `policy_template` tiles.

**Comment by sorantis** [comment]
> IIRC from our discussions we decided to always show the base package tile, plus all the `policy_template` tiles.
we should probably discuss this again, since for integrations with multiple policy_templates (like this Kubernetes example) it can be confusing for the user to understand the difference.

**Comment by ChrsMark** [comment]
@kaiyan-sheng @sorantis I'm opening this for review since changes in Kibana have landed. I tested again and from my view the UI seems to be as expected. Putting some updated screenshots below:

Screenshot 2021-06-16 at 5 01 03 PM

Screenshot 2021-06-16 at 5 01 17 PM

Screenshot 2021-06-16 at 5 01 36 PM

Screenshot 2021-06-16 at 5 02 18 PM

**Comment by ChrsMark** [comment]
I tested the integration manually in a local k8s cluster. Initially added only kube-state-metrics tile and metrics are collected properly. Then I tried to add another tile, kubelet, but I got the following:

Screenshot 2021-06-17 at 4 16 12 PM

Is this expected @jen-huang @sorantis ? How I'm supposed to enable more data_streams to the integration?

Other than that the package/integration itself looks ok so maybe we can proceed and merge it? What do you think @kaiyan-sheng ?

**Comment by jen-huang** [comment]
@ChrsMark What happens if you change the name of the k8 policy (in step 2)? It sounds like it is autofilling it with a name that was already used by the first k8 policy, can you check if that's the case? If so, that might be a bug.

You can also enable more streams by editing the first k8 policy.

**Comment by kaiyan-sheng** [comment]
@ChrsMark I was able to add additional data streams in AWS package. For example: I added ec2 metrics first into aws-1 and then added elb metrics into aws-2: 
Screen Shot 2021-06-17 at 2 51 49 PM

These two are added into the policy as two separate inputs. One for collecting EC2 metrics and another one collecting ELB metrics. The policy looks like this:
```
id: f5ce3420-cf7f-11eb-a518-f7d4117bebd4
revision: 12
outputs:
  default:
    type: elasticsearch
    hosts:
      - 'http://localhost:9200'
output_permissions:
  default:
    aws-1:
      indices:
        - names:
            - metrics-aws.ec2_metrics-default
          privileges:
            - auto_configure
            - create_doc
    aws-2:
      indices:
        - names:
            - metrics-aws.elb_metrics-default
          privileges:
            - auto_configure
            - create_doc
    _elastic_agent_checks:
      cluster:
        - monitor
      indices:
        - names:
            - logs-elastic_agent.*-default
            - metrics-elastic_agent.*-default
          privileges:
            - auto_configure
            - create_doc
agent:
  monitoring:
    enabled: true
    use_output: default
    namespace: default
    logs: true
    metrics: true
inputs:
  - id: b46ec663-1743-47f1-b8d8-03c90a0c38e1
    name: aws-1
    revision: 1
    type: aws/metrics
    use_output: default
    meta:
      package:
        name: aws
        version: 0.6.2
    data_stream:
      namespace: default
    streams:
      - id: aws/metrics-aws.ec2_metrics-b46ec663-1743-47f1-b8d8-03c90a0c38e1
        data_stream:
          dataset: aws.ec2_metrics
          type: metrics
        metricsets:
          - ec2
        period: 5m
        credential_profile_name: elastic-beats
        tags_filter: null
  - id: d44a360d-dc64-4f89-a113-d7bba4d15f14
    name: aws-2
    revision: 1
    type: aws/metrics
    use_output: default
    meta:
      package:
        name: aws
        version: 0.6.2
    data_stream:
      namespace: default
    streams:
      - id: aws/metrics-aws.elb_metrics-d44a360d-dc64-4f89-a113-d7bba4d15f14
        data_stream:
          dataset: aws.elb_metrics
          type: metrics
        metricsets:
          - elb
        period: 1m
        credential_profile_name: elastic-beats
        tags_filter: null
fleet:
  hosts:
    - 'http://localhost:8220'
```

**Comment by ChrsMark** [comment]
@jen-huang I can verify that adding more integrations under the `k8s` policy works fine. Also when I'm trying from integrations page to add extra data_streams then I have to change the name of the integration, so this seems to be the issue:

Screenshot 2021-06-18 at 10 26 59 AM

@jen-huang  I think we might need to improve this experience for our users since they will try the straightforward way as I did and they will fail adding extra data_streams.

**Review by kaiyan-sheng — APPROVED** [review]
Looks good to me!

**Comment by ChrsMark** [comment]
Thanks @kaiyan-sheng !

@sorantis do you think we can proceed with merging this one?

**Comment by jen-huang** [comment]
> when I'm trying from integrations page to add extra data_streams then I have to change the name of the integration

@ChrsMark It's a bug if the name of the integration doesn't automatically increment based on how many other k8 integrations on on the selected policy. Actually, we changed some code here recently (reordered Step 1 & 2), so it might be a regression. I'll take a look.

**Comment by jen-huang** [comment]
> > when I'm trying from integrations page to add extra data_streams then I have to change the name of the integration
> 
> @ChrsMark It's a bug if the name of the integration doesn't automatically increment based on how many other k8 integrations on on the selected policy. Actually, we changed some code here recently (reordered Step 1 & 2), so it might be a regression. I'll take a look.

I just tested this on latest Kibana with system package and found that the name does increment correctly. @ChrsMark do you have any more details around how to reproduce your circumstance? I.e. what is the state of the policy you are adding k8 to (does it already have a k8 integration?), is it the policy that is selected by default in Step 2 or do you have to manually choose it from the dropdown?

**Comment by ChrsMark** [comment]
@jen-huang the name does increment correctly for the default policy but if I try to change the policy before adding the integration the name is not adjusted accordingly so this  the reason for failing. I think that the name should be recalculated every time you pick a different policy from the dropdown list. Make sense?

**Comment by jen-huang** [comment]
@ChrsMark Ah got it, thanks for that clarification. I think this is an existing behavior that we don't have logged as a bug, so I will open something to track that.

**Comment by ChrsMark** [comment]
Thank you @jen-huang! Shall we create a discuss-meta issue to keep track of this one and any other leftovers like keepingVsRemoving the original tile? We can then merge this one since from package perspective works as expected.

**Review by sorantis — APPROVED** [review]

**Comment by sorantis** [comment]
@ChrsMark Is this one going into 7.14? cc @akshay-saraswat

**Comment by ChrsMark** [comment]
Yes!

**Pull request merged**